### ISSUE TITLE
Bug: @typescript-eslint/no-useless-constructor checkParams throws error when param.decorators is undefined

### ISSUE THREAD

**Issue by robsterlini** [typescript-eslint/typescript-eslint#7721]
### Before You File a Bug Report Please Confirm You Have Done The Following...

- [X] I have tried restarting my IDE and the issue persists.
- [X] I have updated to the latest version of the packages.
- [X] I have [searched for related issues](https://github.com/typescript-eslint/typescript-eslint/issues?q=is%3Aissue+label%3A%22package%3A+eslint-plugin%22) and found none that matched my issue.
- [X] I have [read the FAQ](https://typescript-eslint.io/linting/troubleshooting) and my problem is not listed.

### Playground Link

https://typescript-eslint.io/play/#ts=5.2.2&fileType=.tsx&code=MYGwhgzhAECCBGB7ATgFwKLOS6BTAHqrgHYAmMm2y0A3gFDTTCLESrICuwqKAFALa4oYAOa4AXNDbIAlsREBKWg0ZSOAB1zIBQiKNwKA3CsaoAFjIgA6YmEHQAvNADkCFBiwpnxxgF86vkA&eslintrc=N4KABGBEBOCuA2BTAzpAXGUEKQAIBcBPABxQGNoBLY-AWhXkoDt8B6Jge1tmUSWWS0yHJsnxwy%2BDtHRRE0aNMjgwAXxCqgA&tsconfig=N4KABGBEDGD2C2AHAlgGwKYCcDyiAuysAdgM6QBcYoEEkJemy0eAcgK6qoDCAFutAGsylBm3TgwAXxCSgA&tokens=false

It doesn't fail in the playground

### Repro Code

```typescript
class AbortError extends Error {
  constructor(message: string) {
    super(message);
    this.name = 'AbortError';
  }
}
```

### ESLint Config

```javascript
{
  "plugins": ["@nx"],
  "rules": {
    "@typescript-eslint/no-useless-constructor": "error"
  }
}
```

### tsconfig

```jsonc
{
  "compilerOptions": {
        "rootDir": ".",
        "sourceMap": true,
        "declaration": false,
        "moduleResolution": "node",
        "emitDecoratorMetadata": true,
        "experimentalDecorators": true,
        "importHelpers": true,
        "target": "ESNext",
        "module": "ESNext",
        "lib": ["ESNext", "dom"],
        "skipLibCheck": true,
        "skipDefaultLibCheck": true,
        "baseUrl": ".",
  }
}
```

### Expected Result

I should expect the linter to complete and return a success state with no JS errors and no linting violations

### Actual Result

Cannot read properties of undefined (reading 'length')

   Occurred while linting /example.ts:2
   Rule: "@typescript-eslint/no-useless-constructor"

TypeError: Cannot read properties of undefined (reading 'length')
Occurred while linting /example.ts:2
Rule: "@typescript-eslint/no-useless-constructor"
    at /foo/node_modules/@typescript-eslint/eslint-plugin/src/rules/no-useless-constructor.ts:40:24
    at Array.some (<anonymous>)
    at checkParams (/foo/node_modules/@typescript-eslint/eslint-plugin/src/rules/no-useless-constructor.ts:37:29)
    at MethodDefinition (/foo/node_modules/@typescript-eslint/eslint-plugin/src/rules/no-useless-constructor.ts:66:11)
    at ruleErrorHandler (/foo/node_modules/eslint/lib/linter/linter.js:1050:28)
    at /foo/node_modules/eslint/lib/linter/safe-emitter.js:45:58
    at Array.forEach (<anonymous>)
    at Object.emit (/foo/node_modules/eslint/lib/linter/safe-emitter.js:45:38)
    at NodeEventGenerator.applySelector (/foo/node_modules/eslint/lib/linter/node-event-generator.js:297:26)
    at NodeEventGenerator.applySelectors (/foo/node_modules/eslint/lib/linter/node-event-generator.js:326:22)

### Additional Info

Changing `param.decorators.length` to `param.decorators?.length` in `checkParams` within the rule `https://github.com/typescript-eslint/typescript-eslint/blob/main/packages/eslint-plugin/src/rules/no-useless-constructor.ts#L40` in my `node_modules` made this work as expected.

I'm unfamiliar with what goes on under the hood with these things, but it looks like it may have once existed https://github.com/typescript-eslint/typescript-eslint/pull/5450/files#r941593887?

**Comment by bradzacher** [comment]
If I had to guess - you're using an old version of our parser with a new version of our plugin.

Hence it doesn't repro on the playground.

You should double check the installed versions and ensure both are kept at the same version.
https://typescript-eslint.io/linting/troubleshooting#how-do-i-check-to-see-what-versions-are-installed
